k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
You think so? It's a body which can be accessed through a file provider, given a FileProvider interface - the memory-backed "deferred" part is only an internal implementation detail, that's why it's called `Provided-` instead of `Deferred-`. It doesn't have a `getFile` method but only `getProviderUri` instead, there is no two concepts to the outside.

The creating classes also don't care about how the file provider does its job, they just pass the interface obtained from the related FileProvider itself around.